test/api/v3/unit/libs/cron.test.js

```diff
@@ -0,0 +1,491 @@
+/* eslint-disable global-require */
+import moment from 'moment';
+import { cron } from '../../../../../website/src/libs/api-v3/cron';
+import { model as User } from '../../../../../website/src/models/user';
+import * as Tasks from '../../../../../website/src/models/task';
+import { clone } from 'lodash';
+import common from '../../../../../common';
+
+// const scoreTask = common.ops.scoreTask;
+
+describe('cron', () => {
+  let user;
+  let tasksByType = {habits: [], dailys: [], todos: [], rewards: []};
+  let daysMissed = 0;
+  let analytics = {
+    track: sinon.spy(),
+  };
+
+  beforeEach(() => {
+    user = new User({
+      auth: {
+        local: {
+          username: 'username',
+          lowerCaseUsername: 'username',
+          email: 'email@email.email',
+          salt: 'salt',
+          hashed_password: 'hashed_password', // eslint-disable-line camelcase
+        },
+      },
+    });
+
+    user._statsComputed = {
+      mp: 10,
+    };
+  });
+
+  it('updates user.auth.timestamps.loggedin and lastCron', () => {
+    let now = new Date();
+
+    cron({user, tasksByType, daysMissed, analytics, now});
+
+    expect(user.auth.timestamps.loggedin).to.equal(now);
+    expect(user.lastCron).to.equal(now);
+  });
+
+  it('updates user.preferences.timezoneOffsetAtLastCron', () => {
+    let timezoneOffsetFromUserPrefs = 1;
+
+    cron({user, tasksByType, daysMissed, analytics, timezoneOffsetFromUserPrefs});
+
+    expect(user.preferences.timezoneOffsetAtLastCron).to.equal(timezoneOffsetFromUserPrefs);
+  });
+
+  it('resets user.items.lastDrop.count', () => {
+    user.items.lastDrop.count = 4;
+    cron({user, tasksByType, daysMissed, analytics});
+    expect(user.items.lastDrop.count).to.equal(0);
+  });
+
+  describe('end of the month perks', () => {
```

Comment:
Probably should be a test that this stuff doesn't happen when user is not a subscriber.